freqtrade/rpc/telegram.py

```diff
@@ -1517,6 +1520,47 @@ def _show_config(self, update: Update, context: CallbackContext) -> None:
             f"*Current state:* `{val['state']}`"
         )
 
+    @authorized_only
+    def _list_kvals(self, update: Update, context: CallbackContext) -> None:
+        """
+        Handler for /list_kvals <id> <key>.
+        List keyvalues for specified trade (and key if supplied).
+        :param bot: telegram bot
+        :param update: message update
+        :return: None
+        """
+        try:
+            if not context.args or len(context.args) == 0:
+                raise RPCException("Trade-id not set.")
+            trade_id = int(context.args[0])
+            key = None if len(context.args) < 2 else str(context.args[1])
+
+            results = self._rpc._rpc_list_kvals(trade_id, key)
+            messages = []
+            if len(results) > 0:
+                messages = ['Found key-value pair' + 's:  \n' if key is None else ':  \n']
+                for result in results:
+                    lines = [
+                        f"*Key:* `{result['kv_key']}`",
+                        f"*ID:* `{result['id']}`",
+                        f"*Trade ID:* `{result['ft_trade_id']}`",
+                        f"*Type:* `{result['kv_type']}`",
+                        f"*Value:* `{result['kv_value']}`",
+                        f"*Create Date:* `{result['created_at']}`",
+                        f"*Update Date:* `{result['updated_at']}`"
+                    ]
+                    # Filter empty lines using list-comprehension
+                    messages.append("\n".join([line for line in lines if line]))
+                for msg in messages:
+                    self._send_msg(msg)
```

Comment:
You will have to be careful with message length.

as you don't know what the content may eventually be - you could end up with 4000+ characters in one line - which is the limit per message iirc.
Now that would not crash the bot - but simply not send the message - but maybe we can handle that correctly/properly?

Reply:
either way works for me (even splitting in theory - but the result could not be what the user wants depending on the content).
i'm just pointing it out (the message limit is available as a constant in the telegram file).

Reply:
Addressed through https://github.com/freqtrade/freqtrade/pull/6908/commits/8494bea64f0b7534553b28c15bf1b5a1a791e383.